charger/ocpp.go

```diff
@@ -167,146 +175,140 @@ func NewOCPP(id string, connector int, idtag string,
-							meterSampleInterval = time.Duration(val) * time.Second
-						}
-
-					case ocpp.KeyConnectorSwitch3to1PhaseSupported:
-						var val bool
-						if val, err = strconv.ParseBool(*opt.Value); err == nil {
-							c.phaseSwitching = val
-						}
-
-					case ocpp.KeyAlfenPlugAndChargeIdentifier:
-						if c.idtag == defaultIdTag {
-							c.idtag = *opt.Value
-							c.log.DEBUG.Printf("overriding default `idTag` with Alfen-specific value: %s", c.idtag)
-						}
-
-					case ocpp.KeyChargingScheduleAllowedChargingRateUnit:
-						if *opt.Value == "W" || *opt.Value == "Power" {
-							c.chargingRateUnit = types.ChargingRateUnitWatts
-						}
+				case ocpp.KeyMaxChargingProfilesInstalled:
+					if val, err := strconv.Atoi(*opt.Value); err == nil {
+						c.chargingProfileId = val
 					}
 
-					if err != nil {
-						break
+				case ocpp.KeyMeterValuesSampledDataMaxLength:
+					if val, err := strconv.Atoi(*opt.Value); err == nil {
+						meterValuesSampledDataMaxLength = val
+					}
+
+				case ocpp.KeyNumberOfConnectors:
+					var val int
+					if val, err = strconv.Atoi(*opt.Value); err == nil && connector > val {
+						err = fmt.Errorf("connector %d exceeds max available connectors: %d", connector, val)
+					}
+
+				case ocpp.KeySupportedFeatureProfiles:
+					if !c.hasProperty(*opt.Value, smartcharging.ProfileName) {
+						err = fmt.Errorf("the mandatory SmartCharging profile is not supported")
+					}
+					c.hasRemoteTriggerFeature = c.hasProperty(*opt.Value, remotetrigger.ProfileName)
+
+				// vendor-specific keys
+				case ocpp.KeyAlfenPlugAndChargeIdentifier:
+					if c.idtag == defaultIdTag {
+						c.idtag = *opt.Value
+						c.log.DEBUG.Printf("overriding default `idTag` with Alfen-specific value: %s", c.idtag)
 					}
 				}
+
+				if err != nil {
+					break
+				}
 			}
+		}
 
-			rc <- err
-		}, nil)
+		rc <- err
+	}, nil)
 
-		if err := c.wait(err, rc); err != nil {
-			return nil, err
-		}
+	if err := c.wait(err, rc); err != nil {
+		return nil, err
 	}
 
-	if meterValues != "" && meterValues != c.meterValuesSample {
-		if err := c.configure(ocpp.KeyMeterValuesSampledData, meterValues); err != nil {
-			return nil, err
+	// see who's there
+	if c.hasRemoteTriggerFeature {
+		if err := conn.TriggerMessageRequest(core.BootNotificationFeatureName); err == nil {
+			select {
+			case <-time.After(timeout):
+				c.log.DEBUG.Printf("BootNotification timeout")
+			case res := <-cp.BootNotificationRequest():
+				if res != nil {
+					c.bootNotification = res
+				}
+			}
 		}
+	}
 
-		// configuration activated
-		c.meterValuesSample = meterValues
+	// autodetect measurands
+	if meterValues == "" && meterValuesSampledDataMaxLength > 0 {
+		sampledMeasurands := c.tryMeasurands(desiredMeasurands, ocpp.KeyMeterValuesSampledData)
+		if len(sampledMeasurands) > meterValuesSampledDataMaxLength {
+			meterValues = strings.Join(sampledMeasurands[:meterValuesSampledDataMaxLength], ",")
```

Comment:
@premultiply hier verlassen wir uns darauf, dass die Box Leistung an Position 1 schickt. Das ist vielleicht ein bisschen optimistisch?

Reply:
Nein, wie kommst du darauf?

Reply:
Per Default ist meterValuesSampledDataMaxLength 1. Auch wenn es mehr sind würden wir die ersten N nehmen. Das sind aber nicht unbedingt diejenigen die wir haben wollen?

Reply:
Doch, das sind in `desiredMeasurements` in der Reihenfolge der Priorität die abonnierten Messwerte.
Im dümmsten Fall wäre hier MaxLen wirklich 1. Das ist korrekt und da kann man nix machen.
In der Regel ist es aber deutlich höher und fast immer kann man alle gewünschten und verfügbaren Messwerte abonnieren.

Reply:
Ahh- dachte die Liste käme vom Charger 👍🏻

Reply:
Schön wär's...
Aber das wäre ja zu einfach. 😅

Reply:
Die vereinfachte Funktion ist mometan so leider kaputt.
Es werden jetzt nur noch `meterValues` gesetzt wenn es mehr sampledMeasurands als meterValuesSampledDataMaxLength gibt?

Richtig war:
`meterValues` wird auf `meterValuesSampledDataMaxLength` _begrenzt_.

Reply:
Autsch. Korrigiere ich.

Reply:
Fixed by https://github.com/evcc-io/evcc/commit/8eb65b8aae8518e53818d75aea9d8e7abc0e2160